azurerm/resource_arm_stream_analytics_reference_input_blob.go

```diff
@@ -0,0 +1,243 @@
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, jobName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Stream Analytics Stream Input %q (Job %q / Resource Group %q): %s", name, jobName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_stream_analytics_stream_input_blob", *existing.ID)
+		}
+	}
+
+	containerName := d.Get("storage_container_name").(string)
+	dateFormat := d.Get("date_format").(string)
+	pathPattern := d.Get("path_pattern").(string)
+	storageAccountKey := d.Get("storage_account_key").(string)
+	storageAccountName := d.Get("storage_account_name").(string)
+	timeFormat := d.Get("time_format").(string)
+
+	serializationRaw := d.Get("serialization").([]interface{})
+	serialization, err := azure.ExpandStreamAnalyticsStreamInputSerialization(serializationRaw)
+	if err != nil {
+		return fmt.Errorf("Error expanding `serialization`: %+v", err)
+	}
+
+	props := streamanalytics.Input{
+		Name: utils.String(name),
+		Properties: &streamanalytics.ReferenceInputProperties{
+			Type: streamanalytics.TypeReference,
+			Datasource: &streamanalytics.BlobReferenceInputDataSource{
+				Type: streamanalytics.TypeBasicReferenceInputDataSourceTypeMicrosoftStorageBlob,
+				BlobReferenceInputDataSourceProperties: &streamanalytics.BlobReferenceInputDataSourceProperties{
+					Container:   utils.String(containerName),
+					DateFormat:  utils.String(dateFormat),
+					PathPattern: utils.String(pathPattern),
+					TimeFormat:  utils.String(timeFormat),
+					StorageAccounts: &[]streamanalytics.StorageAccount{
+						{
+							AccountName: utils.String(storageAccountName),
+							AccountKey:  utils.String(storageAccountKey),
+						},
+					},
+				},
+			},
+			Serialization: serialization,
+		},
+	}
+
+	if d.IsNewResource() {
+		if _, err := client.CreateOrReplace(ctx, props, resourceGroup, jobName, name, "", ""); err != nil {
+			return fmt.Errorf("Error Creating Stream Analytics Reference Input Blob %q (Job %q / Resource Group %q): %+v", name, jobName, resourceGroup, err)
+		}
+
+		read, err := client.Get(ctx, resourceGroup, jobName, name)
+		if err != nil {
+			return fmt.Errorf("Error retrieving Stream Analytics Reference Input Blob %q (Job %q / Resource Group %q): %+v", name, jobName, resourceGroup, err)
+		}
+		if read.ID == nil {
+			return fmt.Errorf("Cannot read ID of Stream Analytics Reference Input Blob %q (Job %q / Resource Group %q)", name, jobName, resourceGroup)
+		}
+
+		d.SetId(*read.ID)
+	} else {
+		if _, err := client.Update(ctx, props, resourceGroup, jobName, name, ""); err != nil {
+			return fmt.Errorf("Error Updating Stream Analytics Reference Input Blob %q (Job %q / Resource Group %q): %+v", name, jobName, resourceGroup, err)
+		}
+	}
```

Comment:
Given there is vastly different logic for create & update could we maybe split this function into a create and update?